support/k8s/k8s-workload-registrar/README.md

```diff
@@ -22,49 +22,50 @@ The configuration file is a **required** by the registrar. It contains
 
 | Key                        | Type    | Required? | Description                              | Default |
 | -------------------------- | --------| ---------| ----------------------------------------- | ------- |
-| `log_level`                | string  | required | Log level (one of `"panic"`,`"fatal"`,`"error"`,`"warn"`, `"warning"`,`"info"`,`"debug"`,`"trace"`) | `"info"` |
+| `log_level`                | string  | optional | Log level (one of `"panic"`,`"fatal"`,`"error"`,`"warn"`, `"warning"`,`"info"`,`"debug"`,`"trace"`) | `"info"` |
 | `log_path`                 | string  | optional | Path on disk to write the log | |
-| `addr`                     | string  | required | Address to bind the HTTPS listener to | `":8443"` |
-| `cert_path`                | string  | required | Path on disk to the PEM-encoded server TLS certificate | `"cert.pem"` |
-| `key_path`                 | string  | required | Path on disk to the PEM-encoded server TLS key |  `"key.pem"` |
-| `cacert_path`              | string  | required | Path on disk to the CA certificate used to verify the client (i.e. API server) | `"cacert.pem"` |
-| `insecure_skip_client_verification` | boolean | required | If true, skips client certificate verification (in which case `cacert_path` is ignored). See [Security Considerations](#security-considerations) for more details. | `false` |
+| `metrics_addr`             | string  | optional | Address to expose metrics on, use `0` to disable. | 0 |
+| `agent_socket_path`        | string  | optional | Path to the Unix domain socket of the SPIRE agent. If not specified, attempt to connect to the server insecurely | |
```

Comment:
The server addr can contain a unix socket, in which case the agent socket isn't required. I've rewritten it so that a server address starting `unix://` is handled differently, and providing a remote address without an agent socket is an error.

Reply:
I recognize the back-compat code in the server socket path handling now. I think the phrase "connect to the server insecurely" is not accurate in the case where we are connecting to the server over UDS. We might phrase it as "If not specified, the server will be contacted over UDS". IMO, we should fail initialization if the server address is not to a UDS and this option is unset.